src/core/server/saved_objects/saved_objects_service.ts

```diff
@@ -348,6 +359,15 @@ export class SavedObjectsService
         this.typeRegistry.registerType(type);
       },
       getImportExportObjectLimit: () => this.config!.maxImportExportSize,
+      setRepositoryFactoryProvider: (repositoryProvider) => {
+        if (this.started) {
+          throw new Error('cannot call `setRepositoryFactoryProvider` after service startup.');
+        }
+        if (this.respositoryFactoryProvider) {
+          throw new Error('custom repository factory is already set, and can only be set once');
```

Comment:
`set` to `registered`?

Reply:
I changed the function name to `setRepositoryFactoryProvider` to be consistent as setting client factory provider function.